src/main/java/Duke.java

```diff
@@ -1,10 +1,165 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+
 public class Duke {
+
     public static void main(String[] args) {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+
+        Scanner sc = new Scanner(System.in);
+        boolean DukeContinues = true;
+        ArrayList<Task> taskList = new ArrayList<>();
+        String filePath = "data/duke.txt";
+        File file = new File(filePath);
+        while (DukeContinues) {
+            String s = sc.nextLine();
+            String command = "";
+            String restOfString = "";
+            boolean taskListChanged = false;
+            if (s.contains(" ")) {
+                command = s.substring(0, s.indexOf(" "));
+                restOfString = s.substring(s.indexOf(" ") + 1);
+            } else {
+                command = s;
+            }
+            try {
+                switch (command) {
+                    case "bye" -> {
+                        Duke.bye();
+                        DukeContinues = false;
+                    }
+                    case "list" -> Duke.list(taskList);
+                    case "mark" -> {
+                        Duke.mark(taskList, restOfString);
+                        taskListChanged = true;
+                    }
+                    case "unmark" -> {
+                        Duke.unmark(taskList, restOfString);
+                        taskListChanged = true;
+                    }
+                    case "todo" -> {
+                        Duke.addTodo(taskList, restOfString);
+                        taskListChanged = true;
+                    }
+                    case "deadline" -> {
+                        Duke.addDeadline(taskList, restOfString);
+                        taskListChanged = true;
+                    }
+                    case "event" -> {
+                        Duke.addEvent(taskList, restOfString);
+                        taskListChanged = true;
+                    }
+                    case "delete" -> {
+                        Duke.delete(taskList, restOfString);
+                        taskListChanged = true;
+                    }
+                    default -> throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
```

Comment:
According to the [CS2103T coding standard](https://se-education.org/guides/conventions/java/intermediate.html) switch statements should not have indentation for the `case` clauses. Slight amendment needed here 👍